oak_crypto/src/encryptor.rs

```diff
@@ -230,13 +230,15 @@ impl ServerEncryptor {
 /// be multiple responses per request and multiple requests per response.
 // TODO(#4311): Merge `AsyncServerEncryptor` and `ServerEncryptor` once there is `async` support in
 // the Restricted Kernel.
-pub struct AsyncServerEncryptor {
-    recipient_context_generator: Arc<dyn AsyncRecipientContextGenerator>,
+pub struct AsyncServerEncryptor<'a> {
+    recipient_context_generator: &'a (dyn AsyncRecipientContextGenerator + Send + Sync),
```

Comment:
Is there a reason for using a trait object rather than generics? As far as I can tell it will always only be one type for each binary.